plugins/queue/scheduler.js

```diff
@@ -0,0 +1,614 @@
+'use strict';
+
+const logger = require('screwdriver-logger');
+const { merge, reach } = require('@hapi/hoek');
+const Resque = require('node-resque');
+const cron = require('./utils/cron');
+const helper = require('../helper');
+const { timeOutOfWindows } = require('./utils/freezeWindows');
+const DEFAULT_BUILD_TIMEOUT = 90;
+const RETRY_LIMIT = 3;
+const RETRY_DELAY = 5;
+const EXPIRE_TIME = 1800; // 30 mins
+
+/**
+ * Posts a new build event to the API
+ * @method postBuildEvent
+ * @param {Object} config           Configuration
+ * @param {Number} [config.eventId] Optional Parent event ID (optional)
+ * @param {Object} config.pipeline  Pipeline of the job
+ * @param {Object} config.job       Job object to create periodic builds for
+ * @param {String} config.apiUri    Base URL of the Screwdriver API
+ * @return {Promise}
+ */
+async function postBuildEvent(executor, eventConfig) {
+    try {
+        const { pipeline, job, apiUri, eventId, causeMessage, buildId } = eventConfig;
+        const pipelineId = pipeline.id;
+
+        const token = executor.tokenGen({
+            pipelineId,
+            service: 'queue',
+            jobId: job.id,
+            scmContext: pipeline.scmContext,
+            scope: ['user']
+        });
+
+        const admin = await helper.getPipelineAdmin(token, apiUri, pipelineId, executor.requestRetryStrategy);
```

Comment:
error handling for getPipelineAdmin is wrong https://github.com/screwdriver-cd/queue-service/blob/master/plugins/helper.js#L178-L180